tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php

```diff
@@ -750,16 +750,29 @@ public function test_process_directives_does_not_change_inner_html_in_math() {
 	 */
 	private function evaluate( $directive_value ) {
 		$generate_state = function ( $name ) {
+			$obj       = new stdClass();
+			$obj->prop = $name;
 			return array(
-				'key'    => $name,
-				'nested' => array( 'key' => $name . '-nested' ),
+				'key'       => $name,
+				'nested'    => array( 'key' => $name . '-nested' ),
+				'obj'       => $obj,
+				'arrAccess' => new class() implements ArrayAccess {
+					public function offsetExists( $offset ): bool {
+						return true; }
+					#[\ReturnTypeWillChange]
```

Comment:
@sirreal that test implies adding a this line in order to be compatible with PHP 7.2

It seems that the mixed typing is only available in PHP 8 and above.

Should we keep this test then?

Reply:
I tested this for the class body and it works (without errors, deprecations, or warnings) on php 8.3 and 7.2, let's use this:
```php
					public function offsetExists( $offset ): bool { return true; }
					#[\ReturnTypeWillChange]
					public function offsetGet( $offset ) { return $offset; }
					public function offsetSet( $offset, $value ): void {}
					public function offsetUnset( $offset ): void {}
```

> Can you omit the `bool` type hint as the return value?

That triggers a deprecation notice in recent versions (8.3).

Reply:
I'd like to keep the test, if we support the behavior a test is a good thing 🙂

Reply:
> Can you omit the `bool` type hint as the return value?

Nope, it will trigger:

Fatal error: During inheritance of ArrayAccess: Uncaught Return type of ArrayAccess@anonymous::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

Reply:
This is what I meant based on the usage in other places:
<img width="1219" alt="Screenshot 2024-06-03 at 14 09 44" src="https://github.com/WordPress/wordpress-develop/assets/699132/5a0d9578-4e28-4434-a576-40370ff86353">